src/core/basetypes/MCData.cpp

```diff
@@ -606,30 +636,27 @@ Data * Data::decodedDataUsingEncoding(Encoding encoding)
             cur_token = 0;
             mailmime_part_parse(text, text_length, &cur_token,
                                 mime_encoding, &decoded, &decoded_length);
-            data = Data::dataWithBytes(decoded, (unsigned int) decoded_length);
-            mailmime_decoded_part_free(decoded);
+
+            data = Data::dataWithBytesTakeOwnership(decoded, (unsigned int) decoded_length, decodedPartDeallocator);
             return data;
         }
         case EncodingUUEncode:
         {
-            char * dup_data;
-            size_t decoded_length;
             Data * data;
-            char * current_p;
+            const char * current_p;
             
             data = Data::dataWithCapacity((unsigned int) text_length);
-            
-            dup_data = (char *) malloc(text_length);
-            memcpy(dup_data, text, text_length);
-            
-            current_p = dup_data;
+
+            current_p = text;
             while (1) {
+                char decoded_buf[256];
```

Comment:
Ok. since uu encoding format is not too popular, I guess we can fail if the encoding is wrong (more than 200 bytes / lines, etc). Could you add some comment to explain that uu encoding is usually 45 bytes or less / line?